bundle/internal/tf/codegen/generator/walker.go

```diff
@@ -92,7 +91,12 @@ func (w *walker) walk(block *tfjson.SchemaBlock, name []string) error {
 
 		// Assert the attribute type is always set.
 		if v.AttributeType == cty.NilType {
-			return fmt.Errorf("unexpected nil type for attribute %s", k)
+			// If the attribute is a nested type, the type is an object.
+			if v.AttributeNestedType != nil {
```

Comment:
Workaround needed for app attributes not returning `type` field anymore in TF schema. Checking with @miles if it's expected